### PR TITLE
Fcma partial sim matrix

### PR DESCRIPTION
**Pull request by yidawang**
add partial similarity matrix algorithm in correlation-based classification
add the corresponding test code
rename some methods
improve the example code

### CODE REVIEW

File: examples/fcma/classification.py [hunk 1]
```diff
@@ -27,25 +28,32 @@
 logging.basicConfig(level=logging.INFO, format=format, stream=sys.stdout)
 logger = logging.getLogger(__name__)
 
+def example_of_aggregating_sim_matrix(raw_data, labels):
+    # aggregate the similarity matrix to save memory
+    use_clf = svm.SVC(kernel='precomputed', shrinking=False, C=1)
```
Comment (mihaic) [inline examples/fcma/classification.py]: `use_clf` seems like a Boolean name. How about `svm_clf`?

File: brainiak/fcma/classifier.py [hunk 2]
```diff
@@ -43,6 +43,12 @@ class Classifier(BaseEstimator):
     clf: class
         The classifier used, normally a classifier class of sklearn
 
+    num_processed_voxels: int, default 2000
+        Used for SVM with precomputed kernel,
+        every time only compute correlation between num_process_voxels and
+        the whole brain to aggregate the kernel matrices.
+        This is to better use the memory
```
Comment (yidawang) [inline brainiak/fcma/classifier.py]: yes, computing similarity matrices portion by portion can save the memory so as to handle correlations at a larger scale.

File: brainiak/fcma/classifier.py [hunk 3]
```diff
@@ -57,7 +63,10 @@ class Classifier(BaseEstimator):
         default None
```
Comment (yidawang) [inline brainiak/fcma/classifier.py]: deleted this line

File: brainiak/fcma/classifier.py [hunk 4]
```diff
@@ -69,7 +78,10 @@ class Classifier(BaseEstimator):
...
         test_data\_ is set after a prediction is called,
         so that the test data does not need to be regenerated in the
-        subsequent operations, e.g. getting decision values of the prediction
+        subsequent operations, e.g. getting decision values of the prediction.
+        test_data\_ may also be set in the fit method
+        if SVM.SVC with precomputed kernel and the test samples are known.
```
Comment (yidawang) [inline brainiak/fcma/classifier.py]: done

File: brainiak/fcma/classifier.py [hunk 5]
```diff
@@ -79,27 +91,43 @@ class Classifier(BaseEstimator):
 
     num_samples_: int
         The number of samples of the training set
+
+    num_digits_: int
+        The number of digit of the first value of the kernel matrix,
```
Comment (yidawang) [inline brainiak/fcma/classifier.py]: done

File: brainiak/fcma/classifier.py [hunk 6]
```diff
@@ -245,6 +303,12 @@ def fit(self, X, y):
             and prepared for correlation computation.
             assuming all elements of X has the same num_voxels value
         y: labels, len(X) equals len(Y)
+        num_training_samples: int, default None
+            the number of samples that used in the training,
```
Comment (yidawang) [inline brainiak/fcma/classifier.py]: done

File: brainiak/fcma/classifier.py [hunk 7]
```diff
@@ -57,7 +63,10 @@ class Classifier(BaseEstimator):
         default None
         training_data\_ is None except clf is SVM.SVC with precomputed kernel,
         in which case training data is needed to compute
-        the similarity vector for each sample to be classified
+        the similarity vector for each sample to be classified.
+        However, if the test samples are also provided during the fit,
```
Comment (yidawang) [inline brainiak/fcma/classifier.py]: done

File: brainiak/fcma/classifier.py [hunk 8]
```diff
@@ -79,27 +91,43 @@ class Classifier(BaseEstimator):
...
+        it will generate the correlation between some voxels and all voxels
+
         Parameters
         ----------
         X: a list of numpy array in shape [num_TRs, num_voxels]
             len(X) is the number of samples
             assuming all elements of X has the same num_voxels value
+        start_voxel: int, default 0
+            the starting voxel id for correlation computation
+        num_voxels_1: int, default None
```
Comment (yidawang) [inline brainiak/fcma/classifier.py]: done

File: brainiak/fcma/classifier.py [hunk 9]
```diff
@@ -245,6 +303,12 @@ def fit(self, X, y):
             and prepared for correlation computation.
             assuming all elements of X has the same num_voxels value
         y: labels, len(X) equals len(Y)
+        num_training_samples: int, default None
+            the number of samples that used in the training,
+            which is set when the similarity matrix is constructed
+            portion by portion so the similarity vectors of the
+            test data have to be computed here.
+            If it is set, only those samples will be used to fit the model
```
Comment (yidawang) [inline brainiak/fcma/classifier.py]: done

File: examples/fcma/classification.py (hunk 1 above)
Comment (yidawang) [inline examples/fcma/classification.py]: done

**Pull request merged**